### PR TITLE
Print annotation in decompiler

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 1]
```diff
@@ -220,8 +227,16 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
         }
         this
 
-      case tree@Term.Ident(name) =>
-        printType(tree.tpe)
+      case tree @ Term.Ident(name) =>
+        tree.tpe match {
+          case Type.SymRef(_, Types.EmptyPrefix()) | Type.TermRef(_, Types.EmptyPrefix()) => this += name
```
Comment (liufengyun) [inline library/src/scala/tasty/util/ShowSourceCode.scala]: It seems `SymRef` subsumes `TermRef`?

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 2]
```diff
@@ -768,13 +800,35 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
     }
 
     def printDefinitionName(sym: Definition): Buffer = sym match {
-      case ValDef(name, _, _) => this += name
+      case ValDef(name, _, _) => this += name += ".type"
```
Comment (liufengyun) [inline library/src/scala/tasty/util/ShowSourceCode.scala]: why this change?

Comment (nicolasstucki) [reply library/src/scala/tasty/util/ShowSourceCode.scala]: I'm trying to figure out how to print `(x: @annot)`, the issue is that it gets expanded to something like `(x: x @annot)`. I was trying to workaround it by printing `(x: x.type @annot)` instead. I will now try to print it as `(x: @annot)` as it was originally.

**Comment by nicolasstucki** [comment]
@liufengyun could you review the changes in the last commit

**Comment by nicolasstucki** [comment]
@liufengyun last 2 commits

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 3]
```diff
@@ -654,11 +654,20 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
...
+            printAnnotation(annot)
+          case tpe @ Type.TypeRef(name, _) if name.endsWith("$") =>
+            printType(tpe)
+            this += ".type"
+          case tpe @ Type.SymRef(ClassDef(name, _, _, _, _), _) if name.endsWith("$") =>
+            printType(tpe)
+            this += ".type"
+          case tpe => printType(tpe)
+        }
+        printTypeAndAnnots(tree.tpe)
```
Comment (liufengyun) [inline library/src/scala/tasty/util/ShowSourceCode.scala]: `tests/run-with-compiler/i3947c.scala` failed due to the change, I cannot figure out why.

Comment (nicolasstucki) [reply library/src/scala/tasty/util/ShowSourceCode.scala]: It was because `scala.runtime.Null$` and `scala.runtime.Nothing$` where handled as modules. Those are now special cases.
